src/mixins/name-request-mixin.ts

```diff
@@ -23,23 +23,11 @@ export default class NameRequestMixin extends Vue {
    * @returns the name request response payload
    */
   async validateNameRequest (nrNumber: string, phone?: string, email?: string): Promise<NrResponseIF> {
-    const nrResponse: NrResponseIF = await LegalServices.fetchNameRequest(nrNumber).catch(error => {
+    const nrResponse: NrResponseIF = await LegalServices.fetchNameRequest(nrNumber, phone, email).catch(error => {
       this.$root.$emit('invalid-name-request', NameRequestStates.NOT_FOUND)
       throw new Error(`Fetch Name Request error: ${error}`)
     })
```

Comment:
Should this handle the different kinds of errors (missing phone/email, invalid phone/email, etc)?

What does the UI show if I search for a NR without phone/email, or with invalid phone/email?

Reply:
Oh, wait, I see you did that in LegalServices! Nice.

But doesn't the error handling above override the error from LegalServices?

Reply:
Yes, I agree
I think we can remove the above error handling.
What do you think?

Reply:
I'm not sure. Why don't you simulate some errors and see what happens?

I think you need to keep the "emit" so that a dialog is displayed.

Reply:
Display like this when we get the errors from API.

<img width="973" alt="Screenshot 2023-11-09 at 7 51 05 AM" src="https://github.com/bcgov/business-edit-ui/assets/54437031/79211373-063b-4f68-9e19-f7db952067cb">

Reply:
The error handling above only handles a single case (not_found).

The error handling you deleted below previously handled incorrect_email and incorrect_phone.

However, the dialog previously displayed the same thing for all 3 cases. Furthermore, you no longer use the incorrect_email or incorrect_phone.

I think you should retain the error case from the API call up to the dialog. To do that, I would move the response handling from fetchNameRequest() to here, and I'd replace incorrect_email and incorrect_phone with new enums. This design would allow the dialog to decide what message to display (and I wouldn't change that) instead of losing the different cases in the underlying logic. Does this make sense?

Reply:
I updated

Reply:
Thanks for the updates. Looks good!